applications/HDF5Application/custom_utilities/container_io_utils.h

```diff
@@ -0,0 +1,477 @@
+        TLSType<TDataType>& rTLS) const
+    {
+        // Since the array_1d<double, 4> and array_1d<double 9> interfaces are not present in
+        // elements and conditions.
+        if constexpr(!std::is_same_v<TDataType, array_1d<double, 4>> && !std::is_same_v<TDataType, array_1d<double, 9>>) {
+
+            using list_traits = DataTypeTraits<std::vector<TDataType>>;
+
+            using vector_traits = DataTypeTraits<Vector<typename list_traits::PrimitiveType>>;
+
+            const_cast<TEntityType&>(rEntity).CalculateOnIntegrationPoints(rVariable, rTLS.mList, mrProcessInfo);
+
+            const auto size = list_traits::Size(rTLS.mList);
+            if (rTLS.mVector.size() != size) {
+                rTLS.mVector.resize(size, false);
+            }
+
+            list_traits::CopyToContiguousData(vector_traits::GetContiguousData(rTLS.mVector), rTLS.mList);
+
+            return rTLS.mVector;
+        } else {
+            KRATOS_ERROR << "The gauss point interface in elements/conditions for Variable<array_1d<double, 4>> and Variable<array_1d<double, 9>> are not defined.";
+            return rTLS.mVector;
+        }
+    }
+
+    template<class TEntityType, class TDataType>
+    inline void SetValue(
+        TEntityType& rEntity,
+        const Variable<TDataType>& rVariable,
+        const ComponentDataType<TDataType>& rValue) const
+    {
+        KRATOS_ERROR << "GaussPointValueIO does not support setting values from vertices";
+    }
+
+    ///@}
+
+private:
+    ///@name Private member variables
+    ///@{
+
+    const ProcessInfo& mrProcessInfo;
+
+    ///@}
+};
+
+template<class TContainerIOType>
+std::string GetContainerIOType()
+{
+    if constexpr(std::is_same_v<TContainerIOType, HistoricalIO>) {
+        return "HISTORICAL";
+    } else if constexpr(std::is_same_v<TContainerIOType, NonHistoricalIO>) {
+        return "NONHISTORICAL";
+    } else if constexpr(std::is_same_v<TContainerIOType, FlagIO>) {
+        return "FLAGS";
+    } else if constexpr(std::is_same_v<TContainerIOType, BossakIO>) {
+        return "HISTORICAL_BOSSAK";
+    } else if constexpr(std::is_same_v<TContainerIOType, VertexValueIO>) {
+        return "INTERPOLATED";
+    } else if constexpr(std::is_same_v<TContainerIOType, GaussPointValueIO>) {
+        return "GAUSS_POINT";
+    } else {
+        static_assert(!std::is_same_v<TContainerIOType, TContainerIOType>, "Unsupported container io type.");
+    }
+}
+
+template<class TContainerType>
+std::string GetContainerType()
+{
+    if constexpr(std::is_same_v<TContainerType, ModelPart::NodesContainerType>) {
+        return "NODES";
+    } else if constexpr(std::is_same_v<TContainerType, ModelPart::ConditionsContainerType>) {
+        return "CONDITIONS";
+    } else if constexpr(std::is_same_v<TContainerType, ModelPart::ElementsContainerType>) {
+        return "ELEMENTS";
+    } else if constexpr(std::is_same_v<TContainerType, Detail::VertexContainerType>) {
+        return "VERTICES";
```

Comment:
Nodes would be more consistent

Reply:
As @matekelemen has told, these are not exactly mesh nodes, these are some random points in the mesh used for line outputs.